src/modules/roc_sndio/target_sox/roc_sndio/sox_backend.cpp

```diff
@@ -202,47 +184,15 @@ void SoxBackend::set_frame_size(core::nanoseconds_t frame_length,
     sox_get_globals()->bufsiz = size * sizeof(sox_sample_t);
 }
 
-bool SoxBackend::probe(const char* driver, const char* inout, int filter_flags) {
-    core::Mutex::Lock lock(mutex_);
-
-    driver = map_to_sox_driver(driver);
-
-    if (!select_defaults(driver, inout)) {
-        return false;
-    }
-
-    const sox_format_handler_t* handler = sox_write_handler(inout, driver, NULL);
-    if (!handler) {
-        return false;
-    }
-
-    if (handler->flags & SOX_FILE_DEVICE) {
-        if ((filter_flags & FilterDevice) == 0) {
-            return false;
-        }
-    } else {
-        if ((filter_flags & FilterFile) == 0) {
-            return false;
-        }
-    }
-
-    return true;
-}
-
 ISink* SoxBackend::open_sink(core::IAllocator& allocator,
                              const char* driver,
                              const char* output,
-                             const Config& config) {
+                             const Config& config,
+                             int filter_flags) {
     core::Mutex::Lock lock(mutex_);
-
     first_created_ = true;
 
     driver = map_to_sox_driver(driver);
-
-    if (!select_defaults(driver, output)) {
-        return NULL;
-    }
-
     core::ScopedPtr<SoxSink> sink(new (allocator) SoxSink(allocator, config), allocator);
```

Comment:
It's not very good that we're reusing sink variable and call open() multiple times. It's too easy to make a mistake in sink implementation and leave the sink in some invalid state after a failed open, which would affect next open call. So better re-create sink each time we try open.